src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.cs

```diff
@@ -205,11 +205,17 @@ private void EnsureWriteCapacity(int pendingCount)
                 throw new OverflowException();
             }
 
-            if (_buffer is null || _buffer.Length - _offset < pendingCount)
+            int currentCapacity = _buffer is null ? 0 : _buffer.Length;
+            int requiredCapacity = _offset + pendingCount;
+            if (currentCapacity < requiredCapacity)
             {
-                const int BlockSize = 1024;
-                int blocks = checked(_offset + pendingCount + (BlockSize - 1)) / BlockSize;
-                Array.Resize(ref _buffer, BlockSize * blocks);
+                int newCapacity = currentCapacity == 0 ? 1024 : currentCapacity * 2;
+                if ((uint)newCapacity > (uint)Array.MaxLength || newCapacity < requiredCapacity)
+                    newCapacity = requiredCapacity;
```

Comment:
Braces are always appreciated :)

```suggestion
                {
                    newCapacity = requiredCapacity;
                }
```